src/fem/fe_system.cc

```diff
@@ -0,0 +1,416 @@
+/*!
+ *
+﻿ * Copyright (C) 2015 Technical University of Liberec.  All rights reserved.
+ * 
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 3 as published by the
+ * Free Software Foundation. (http://www.gnu.org/licenses/gpl-3.0.en.html)
+ * 
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+ *
+ * 
+ * @file    fe_system.cc
+ * @brief   Class FESystem for compound finite elements.
+ * @author  Jan Stebel
+ */
+
+#include "fem/fe_system.hh"
+#include "system/global_defs.h"
+#include "quadrature/quadrature.hh"
+
+using namespace std;
+
+
+
+// template<unsigned int dim,unsigned int spacedim>
+// std::vector<FiniteElement<dim,spacedim> > expand_fe_vector(const std::vector<std::pair<FiniteElement<dim,spacedim>, unsigned int> > &fe)
+// {
+//   std::vector<FiniteElement<dim,spacedim> > finite_elements;
+//   
+//   for (auto fe_pair : fe)
+//     for (unsigned int i=0; i<fe_pair.second; i++)
+//       finite_elements.push_back(fe_pair.first);
+// 
+//   return finite_elements;
+// }
+
+
+
+template<unsigned int dim, unsigned int spacedim>
+FESystem<dim,spacedim>::FESystem(FiniteElement<dim,spacedim> *fe, unsigned int n)
+{
+  FiniteElement<dim,spacedim>::init(0, false);
+  std::shared_ptr<FiniteElement<dim,spacedim> > fe_ptr(fe);
```

Comment:
1. We should make a copy of *fe, which should be a reference.
2. We should support system of different finite elements, e.g constructor:
FeSystem(vector<pair<FiniteElement, uint>> fe_powers)

or
FeSystem(vector<FiniteElement> fes, vector<uint> multiplicity = {})

Reply:
Two constructors:
FESystem(shared_ptr<FiniteElement...> fe, quantity=Vector) // Vector , Tensor ... determines number of 
FESystem(shared_ptr<FiniteElement...> fe, n) // Mixed system 
FESystem(vector<shared_ptr<FiniteElement...>> fe_vec)